docs/articles/commands/text/custom_prefix_handler.md

```diff
@@ -0,0 +1,97 @@
+---
+uid: articles.commands.custom_prefix_handler
+title: Custom Prefix Handler
+---
+
+# Adding dynamic prefixes
+
+Prefixes are commonplace among Discord bots, and are used to determine if a message is a command or not. By default, DSharpPlus uses the `!` prefix, but you can change this to whatever you want. However, what if you want to have different prefixes for different servers? This is where custom prefix handlers come in.
+
+There are two manners of going about this; you can either use a prefix resolver delegate, or implement the `IPrefixResolver` interface. The former is simpler, but the latter is more powerful.
+
+# [Prefix resolver delegate](#tab/prefix_resolver_delegate)
+The prefix resolver delegate is very simple to use. Any method that can be converted into a `Func<CommandsExtension, DiscordMessage, ValueTask<int>>` (a method that takes a `CommandsExtension` and a `DiscordMessage`, and returns a `ValueTask<int>`) can be used as a prefix resolver delegate. This method will be called for every message, and the return value will be used as the prefix length. If the return value is -1, the message will be ignored.
```

Comment:
it is probably worth calling out that the delegate isn't actually a `Func<CommandsExtension, DiscordMessage, ValueTask<int>>` and that binding to Func with whatever variation of reflection will fail

Reply:
I'd wager that users doing such tomfoolery deserve the runtime exploding on them (and also, should use IPrefixResolver if they're going through the trouble) -- while technically PrefixResolverDelegateAsync, its signature is effectively the func above, and the compiler is OK with converting the method group with such a signature to the delegate, so for most intents and purposes I think it's fine, unless there's some common case in which this case could be problematic?

Reply:
doesn't cost us anything to call it out. or alternatively make the delegate a func, which is arguably better anyways